### PR TITLE
Frontpage: add communities carousel [REQUIRES inveniosoftware/invenio-theme#343 invneiosoftware/invenio-communities#844]

### PR DESCRIPTION
**Pull request by Pineirin**
closes: #9
requires: inveniosoftware/invenio-theme#343, inveniosoftware/invenio-communities#844

Computer:
Screenshot from 2022-12-08 09-35-56

Tablet:
Screenshot from 2022-12-08 09-36-35

Mobile:
Screenshot from 2022-12-08 09-36-45

### CODE REVIEW

File: assets/js/invenio_communities/community/communities-carousel/overrides/CommunitiesCarousel.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    this.cancellableFetch = withCancel(http.get(fetchUrl));
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits });
+      this.setCarouselTimer();
+    } catch (error) {
+      console.error(error);
+    }
+  };
```
Comment (kpsherva) [inline assets/js/invenio_communities/community/communities-carousel/overrides/CommunitiesCarousel.js:90]: this part does not need to be overridden, only the layout in the render function

File: assets/js/invenio_communities/community/communities-carousel/overrides/CarouselItem.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React from "react";
+import PropTypes from "prop-types";
+import _truncate from "lodash/truncate";
+import { Image } from "react-invenio-forms";
+import { Grid, Header, Item } from "semantic-ui-react";
+
+export const overriddenCarouselItem = ({community, defaultLogo, className}) => {
```
Comment (kpsherva) [inline assets/js/invenio_communities/community/communities-carousel/overrides/CarouselItem.js:15]:
```suggestion
export const CdsCarouselItem = ({community, defaultLogo, className}) => {
```
classes' names should be starting with a capital letter

**Comment by kpsherva** [comment]
could you add a responsive version of the screenshot?

File: assets/less/cds-rdm/globals/site.variables [hunk 3]
```diff
@@ -48,10 +48,11 @@
 @placeholderTextColor: #abadb1;
 
 @recordsListTitleBackground: #313741;
-@recordsListMetadataBackground: #d6dde8;
+@metadataBackground: #d6dde8;
```
Comment (kpsherva) [inline assets/less/cds-rdm/globals/site.variables:51]: why not to use the previous variable name ? did something change in the base theme ?

File: assets/less/cds-rdm/globals/site.overrides [hunk 4]
```diff
@@ -88,15 +88,93 @@
...
+      }
+    }
+  }
+
+  .ui.items {
+
+    & > .item {
+
+      position:relative;
+      border-style: inset !important;
```
Comment (kpsherva) [inline assets/less/cds-rdm/globals/site.overrides:137]: this style causes a 3D effect around the item, which was not present on the mockup. please use the default border style without inset.

File: assets/less/cds-rdm/globals/site.variables (hunk 3 above)
Comment (Pineirin) [reply assets/less/cds-rdm/globals/site.variables:51]: Previously this background color was only used by the recordsList whilst now, it isn't.

**Review by kpsherva — APPROVED** [review]
LGTM! Should we demo it?

**Comment by kpsherva** [comment]
Updated screenshots:
Screenshot 2022-12-20 at 14 35 13
Screenshot 2022-12-20 at 14 35 49
Screenshot 2022-12-20 at 14 37 51

**Review by zzacharo — APPROVED** [review]

**Pull request merged**